src/main/java/com/dev/cinema/validation/EmailValueValidator.java

```diff
@@ -0,0 +1,15 @@
+package com.dev.cinema.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValueValidator implements ConstraintValidator<EmailValueConstraint, String> {
+    private static final String EMAIL_REGEX = "^[\\w!#$%&'*+/=?`{|}~^-"
+            + "]+(?:\\.[\\w!#$%&'*+/=?`{|}~^-]+)*"
+            + "@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        return email.matches(EMAIL_REGEX);
```

Comment:
let's add null check - and we can remove @NotNull constraint above email field
the problem is that we can not ensure the order of validation